### ISSUE TITLE
Typescript error TS2367 - This condition will always return 'true' since the types...

### ISSUE THREAD

**Issue by TomasWeg** [stephenh/ts-proto#190]
This typescript error is thrown at compile: 
`This condition will always return 'true' since the types 'Constructor<Long>' and '{ default: LongConstructor; prototype: Long; MAX_UNSIGNED_VALUE: Long; MAX_VALUE: Long; MIN_VALUE: Long; NEG_ONE: Long; ONE: Long; ... 11 more ...; fromValue(val: string | ... 2 more ... | { ...; }): Long; }' have no overlap.`

While generating `google/protobuf/wrappers.proto` at this line:
```typescript
if (util.Long !== Long) { // Exactly this
  util.Long = Long as any;
  configure();
}
```

**Comment by stephenh** [comment]
...oh, you have `esModuleInterop: true` on. 

Unfortunately, generating output that can please some downstream projects using `esModuleInterop` and some _not_ using `esModuleInterop` is tricky.

I'll poke around.

**Comment by stephenh** [comment]
Okay, I added a new `esModuleInterop=true` flag that produces the right `Long` imports. Released in v1.45.0.

Thanks for the report!

**Comment by jordan-bonecutter** [comment]
For those running into this in the future, run `protoc` with `--ts_proto_opt=esModuleInterop=true`
